testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveCoercion.java

```diff
@@ -127,10 +127,14 @@ private static HiveTableDefinition.HiveTableDefinitionBuilder tableDefinitionBui
                         "    longdecimal_to_longdecimal            DECIMAL(20,12)," +
                         //"    float_to_decimal           " + floatType + "," + // this coercion is not permitted in Hive 3. TODO test this on Hive < 3.
                         //"    double_to_decimal          DOUBLE," + // this coercion is not permitted in Hive 3. TODO test this on Hive < 3.
-                        "    decimal_to_float           DECIMAL(10,5)," +
-                        "    decimal_to_double          DECIMAL(10,5)," +
-                        "    varchar_to_bigger_varchar  VARCHAR(3)," +
-                        "    varchar_to_smaller_varchar VARCHAR(3)" +
+                        "    decimal_to_float                   DECIMAL(10,5)," +
```

Comment:
@findinpath suggested to add JSON data type to this class.

Reply:
yes, I know, I tried but it doesn't work I did something like:
```
public static final HiveTableDefinition HIVE_COERCION_JSON = tableDefinitionBuilder("TEXTFILE", Optional.empty(), Optional.of("SERDE 'org.apache.hive.hcatalog.data.JsonSerDe'"))
            .setNoData()
            .build();
```

and use it the same way other types are used but get:
```
Cannot validate serde: org.apache.hive.hcatalog.data.JsonSerDe
```

Reply:
Previous error was due to old hive version. With singlenode-hdp3 there is different error:
```
Caused by: java.sql.SQLException: Error while processing statement: FAILED: Execution Error, return code 1 from org.apache.hadoop.hive.ql.exec.DDLTask. Error: type expected at the position 0 of '<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:smallint:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>:<derived from deserializer>' but '<' is found.
tests               | 	at org.apache.hive.jdbc.HiveStatement.execute(HiveStatement.java:275)
tests               | 	at io.trino.tempto.query.JdbcQueryExecutor.executeQueryNoParams(JdbcQueryExecutor.java:128)
tests               | 	at io.trino.tempto.query.JdbcQueryExecutor.execute(JdbcQueryExecutor.java:112)
tests               | 	... 17 more
tests               | 	Suppressed: java.lang.Exception: Query: ALTER TABLE tempto_mut_textfile_hive_coercion_b0t5xhec CHANGE COLUMN tinyint_to_smallint tinyint_to_smallint smallint
tests               | 		at io.trino.tempto.query.JdbcQueryExecutor.executeQueryNoParams(JdbcQueryExecutor.java:136)
tests               | 		... 18 more
```

these exceptions come from hive operations like:
```
onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN tinyint_to_smallint tinyint_to_smallint smallint", tableName));
```

So I don't think there is a way to add tests for json.

Reply:
😞 OK, let's ignore this for now